### PR TITLE
feature[TW15669]: Rework Viewer Multi-Edit in nebula

### PR DESCRIPTION
**Pull request by jmidvidy**

### CODE REVIEW

**Review by lcaron — DISMISSED** [review]

File: widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/action/ColumnMultiEditAction.java [hunk 1]
```diff
@@ -59,17 +59,16 @@ public void run() {
          XViewerLib.popup(XViewerText.get("error"), XViewerText.get("error.no_columns.multi_editable")); //$NON-NLS-1$ //$NON-NLS-2$
          return;
       }
-      ListDialogSortable ld = new ListDialogSortable(new XViewerColumnSorter(), xViewer.getTree().getShell());
-      ld.setMessage(XViewerText.get("ColumnMultiEditAction.title")); //$NON-NLS-1$
+      String title = XViewerText.get("ColumnMultiEditAction.title"); //$NON-NLS-1$
+      ListDialogSortableFiltered ld = new ListDialogSortableFiltered(title, "Select an attribute.", new PatternFilter(),
```
Comment (lcaron) [inline widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/action/ColumnMultiEditAction.java:63]: Should use XViewerText.get(XX) for internationalisation. Please do not forget both German an French Translation
(for French it is "Sélectionner un attribut")

File: widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+         new String[] {XViewerText.get("button.ok"), XViewerText.get("button.cancel")}, 0); //$NON-NLS-1$ //$NON-NLS-2$
+      this.contentProvider = contentProvider;
+      this.labelProvider = labelProvider;
+      this.patternFilter = patternFilter;
+      this.viewerSorter = viewerSorter;
+      setShellStyle(getShellStyle() | SWT.RESIZE);
+   }
+
+   protected void createPreCustomArea(Composite parent) {
+      // provided for subclass implementation
```
Comment (lcaron) [inline widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java:75]: Like the comment (very useful) 👍

File: widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+         treeViewer.getViewer().setInput(input);
+      }
+   }
+
+   /**
+    * Sets the initial selection. Convenience method.
+    *
+    * @param object the initial selection.
+    */
+   public void setInitialSelections(Collection<? extends Object> initialSelections) {
```
Comment (lcaron) [inline widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java:99]: Why use plural ?
setInitialSelection seems more relevant

File: widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+      treeViewer.getViewer().getTree().setLayoutData(gd);
+      treeViewer.getViewer().setContentProvider(contentProvider);
+      treeViewer.getViewer().setLabelProvider(labelProvider);
+      treeViewer.getViewer().setSorter(viewerSorter);
+      treeViewer.getViewer().setAutoExpandLevel(0);
+      treeViewer.setQuickSelectionMode(true);
+      if (viewerComparator != null) {
+         treeViewer.getViewer().setComparator(viewerComparator);
+      }
+      treeViewer.getViewer().addSelectionChangedListener(new ISelectionChangedListener() {
```
Comment (lcaron) [inline widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java:145]: Please use lambda

File: widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+         @Override
+         public void selectionChanged(SelectionChangedEvent event) {
+            updateSelected();
+            updateStatusLabel();
+         }
+      });
+      final ListDialogSortableFiltered dialog = this;
+      treeViewer.getViewer().addDoubleClickListener(new IDoubleClickListener() {
```
Comment (lcaron) [inline widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java:153]: Please use lambda

File: widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+         @Override
+         public void doubleClick(DoubleClickEvent event) {
+            updateSelected();
+            dialog.okPressed();
+         }
+      });
+      treeViewer.getFilterControl().addModifyListener(new ModifyListener() {
```
Comment (lcaron) [inline widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java:161]: Please use lambda

File: widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+      selected.clear();
+      for (Object obj : ((StructuredSelection) treeViewer.getViewer().getSelection()).toArray()) {
+         selected.add(obj);
+      }
+   }
+
+   protected void updateStatusLabel() {
+      Boolean result = isComplete();
+      if (!result) {
+         statusLabel.setText("Error!");
```
Comment (lcaron) [inline widgets/xviewer/org.eclipse.nebula.widgets.xviewer/src/org/eclipse/nebula/widgets/xviewer/util/internal/dialog/ListDialogSortableFiltered.java:209]: Please use getText for i18n

**Review by lcaron — CHANGES_REQUESTED** [review]
A few comments to use lambda and manage i18n, but it is a great job. Congrats

**Comment by dondunne** [comment]
Aborting this as Jeremy isn't committing to Nebula anymore.  I merged this change into another commit and resolved comments, except the Lambda ones.

**Pull request closed without merging**